envoy/api/v2/route/route.proto

```diff
@@ -542,6 +542,13 @@ message RedirectAction {
   // The path portion of the URL will be swapped with this value.
   string path_redirect = 2;
 
+  // [#not-implemented-hide:]
+  // The path portion of the URL can be rewritten with original path replaced
+  // according to rules of this regex.
+  // This helps to use original path as a part of redirect action when prefix/suffix
+  // rewriting is needed.
```

Comment:
Needs examples of what the syntax for match groups looks like, etc.

Reply:
Thanks for the review @htuch .Pasting example that is mentioned in the envoy issue by Brian Ramos . rewrite rules look something this : rewrite ^/foo/(bar)/(.*)$ https://$server_name/$1/$2 permanent;
TBH: This is not a requirement from my side, picked this up to get hang of codebase.

Reply:
The "permanent" part should be a separate enum field for the type of redirect.

Reply:
@ggreenway ack